ehrapy/api/preprocessing/encoding/_encode.py

```diff
@@ -504,6 +594,99 @@ def _undo_encoding(
             layers={"original": new_x.copy()},
         )
 
+    @staticmethod
+    def _delete_all_encodings(adata: AnnData) -> Tuple[Optional[np.ndarray], Optional[list]]:
+        """Delete all encoded columns and keep track of their indices
+        Args:
+            adata: The AnnData object to operate on
+
+        Returns:
+            A temporary X were all encoded columns are deleted and all var_names of unencoded columns
+        """
+        var_names = list(adata.var_names)
```

Comment:
Just use adata.var_names directly. Wasted line of code imo. Does not make it this much more readable.

Reply:
var_namesw returns an index, not a list iirc

Reply:
var_names returns an index, not a list iirc

Reply:
You are correct: https://anndata.readthedocs.io/en/stable/anndata.AnnData.var_names.html

Sorry.